core/src/main/java/com/netflix/msl/msg/MslControl.java

```diff
@@ -2306,6 +2307,20 @@ public MessageInputStream call() throws MslException, MslErrorResponseException,
                     throw new MslErrorResponseException("Error receiving the message header.", rt, e);
                 }
                 throw e;
+            } catch (final IOException e) {
+                // If we were cancelled then return null.
+                if (cancelled(e)) return null;
+                
+                // Maybe we can send an error response.
+                try {
+                    sendError(ctx, debugCtx, null, null, null, MslError.MSL_COMMS_FAILURE, null, out);
+                } catch (final Throwable rt) {
+                    // If we were cancelled then return null.
+                    if (cancelled(rt)) return null;
+                    
+                    throw new MslErrorResponseException("Error receiving the message header.", rt, e);
```

Comment:
2 questions:
1) If an IOException occurred, is it a good assumption that network communication can still be possible to send a response?

2) Why does the message say "Error receiving the message header." when this exception was a result of sending a response?

Reply:
1. It's a weak assumption. But if I can get a response to the remote entity to provide an error, I want to try. The input and output stream states may not be tightly coupled, depending on the underlying protocols and hardware.
2. The error, as far as the local caller is concerned, is that there was a problem receiving the message header. The use of MslErrorResponseException indicates there was also an error attempting to send an error response (see documentation on MslErrorResponseException class).

Reply:
Maybe a message like - "Error encountered while attempting to send a response as a result of an error that occurred while receiving the message header." ?

Reply:
The text you're proposing is covered by the definition of MslErrorResponseException: "Thrown when an exception occurs while attempting to create and send an automatically generated error response."

The documentation on MslErrorResponseException says "The detail message should describe the error that triggered the automatically generated error response."

If we change the detail message to not say that, we would have to redefine MslErrorResponseException.

Why do you feel it is important for the detail message to include text covering the specific exception type (i.e. class) that was thrown? Does that help something somewhere?

Reply:
I just thought it would be more intuitive to the person who sees the message. Bear in mind that most people haven't read all the documentation and aren't likely to remember all of it even if they do. But it's not that big of a deal, hence I approved the PR. The more important issue was addressing the IOException being thrown properly.